src/shared/codelens/localLambdaRunner.ts

```diff
@@ -247,12 +247,20 @@ export class LocalLambdaRunner {
         await command.execute()
 
         if (this.localInvokeParams.isDebug) {
-            await waitForDebugPort({
+            const isPortOpen: boolean = await waitForDebugPort({
                 debugPort: this.debugPort,
                 configuration: this.configuration,
                 channelLogger: this.channelLogger
             })
 
+            if (!isPortOpen) {
+                this.channelLogger.warn(
+                    'AWS.samcli.local.invoke.port.not.open',
+                    // tslint:disable-next-line:max-line-length
+                    'Unable to detect the debug port. The debugger might not succeed in attaching to your SAM Application.'
```

Comment:
Nit: To me, 'detect' reads as "We tried to determine the right port, but couldn't", not "We know the right port, it just isn't open".